application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,319 @@
+
+        event.getInteraction().getComponent().getOptions().stream().map(selectedOption -> {
+            Role role = guild.getRoleById(selectedOption.getValue());
+
+            if (null == role) {
+                handleNullRole(selectedOption);
+            }
+
+            return role;
+        }).filter(Objects::nonNull).forEach(role -> {
+            if (selectedRoles.contains(role)) {
+                rolesToAdd.add(role);
+            } else {
+                rolesToRemove.add(role);
+            }
+        });
+
+        handleRoleModifications(event, event.getMember(), guild, rolesToAdd, rolesToRemove);
+    }
+
+    /**
+     * Handles the selection of the {@link SelectionMenu} if it came from a builder.
+     *
+     * @param event the <b>unacknowledged</b> {@link ComponentInteraction}
+     * @param selectedRoles the {@link Role roles} selected by the {@link User} from the
+     *        {@link ComponentInteraction} event
+     */
+    private void handleNewRoleBuilderSelection(@NotNull final ComponentInteraction event,
+            final @NotNull Collection<? extends Role> selectedRoles) {
+        SelectionMenu.Builder menu =
+                SelectionMenu.create(generateComponentId(event.getUser().getId()))
+                    .setPlaceholder("Select your roles")
+                    .setMaxValues(selectedRoles.size())
+                    .setMinValues(0);
+
+        selectedRoles.forEach(role -> menu.addOption(role.getName(), role.getId()));
+
+        event.getChannel()
+            .sendMessageEmbeds(event.getMessage().getEmbeds().get(0))
+            .setActionRow(menu.build())
+            .queue();
+
+        event.reply("Message sent successfully!").setEphemeral(true).queue();
+    }
+
+    /**
+     * Logs that the role of the given {@link SelectOption} doesn't exist anymore.
+     *
+     * @param selectedOption the {@link SelectOption}
+     */
+    private static void handleNullRole(final @NotNull SelectOption selectedOption) {
+        logger.info(
+                "The {} ({}) role has been removed but is still an option in the selection menu",
+                selectedOption.getLabel(), selectedOption.getValue());
+    }
+
+    /**
+     * Updates the roles of the given member
+     *
+     * @param event an <b>unacknowledged</b> {@link Interaction} event
+     * @param member the member to update the roles of
+     * @param guild what guild to update the roles in
+     * @param additionRoles the roles to add
+     * @param removalRoles the roles to remove
+     */
+    private static void handleRoleModifications(@NotNull final Interaction event,
+            final Member member, final @NotNull Guild guild, final Collection<Role> additionRoles,
+            final Collection<Role> removalRoles) {
+        guild.modifyMemberRoles(member, additionRoles, removalRoles)
+            .flatMap(empty -> event.reply("Updated your roles!").setEphemeral(true))
```

Comment:
Just to follow a bit of the wording sense that's used in most other messages.
Could this be changed to something like "Your roles have been updated!" ?